File: PascalsTriangle.cpp
class Solution {
public:
	vector<vector<int> > generate(int numRows) {
		vector<vector<int> > res;
		vector<int> a(1,1);
		if (0==numRows)
			return res;
		res.push_back(a);
		if (1==numRows)
			return res;
		a.push_back(1);
		res.push_back(a);
		if (2==numRows)
			return res;

		int numOfEle = 0;
		for (int i = 3; i < numRows+1; ++i) {
			numOfEle = i;
			a = vector<int>(numOfEle,1);
			res.push_back(a);
			for (int j = 2; j < numOfEle; ++j) {
				res[i-1][j-1] = res[i-2][j-2] + res[i-2][j-1];
			}

		}
		return res;
	}
};

File: jiuzhang/pascals-triangle/pascals-triangle.java
/**
 * 本代码由九章算法编辑提供。没有版权欢迎转发。
 * - 九章算法致力于帮助更多中国人找到好的工作，教师团队均来自硅谷和国内的一线大公司在职工程师。
 * - 现有的面试培训课程包括：九章算法班，系统设计班，九章强化班，Java入门与基础算法班，
 * - 更多详情请见官方网站：http://www.jiuzhang.com/
 */

public class Solution {
    public ArrayList> generate(int numRows) {
        ArrayList> rst = new ArrayList<ArrayList<Integer>>();
        if (numRows == 0) {
            return rst;
        }

        ArrayList first = new ArrayList<Integer>();
        first.add(0, 1);
        rst.add(first);

        for (int i = 1; i < numRows; i++) {
            ArrayList tmp = new ArrayList<Integer>(i + 1);
            for (int j = 0; j < i + 1; j++){
                tmp.add(-1);
            }
            ArrayList prev = rst.get(i - 1);
            tmp.set(0, prev.get(0));
            tmp.set(i, prev.get(i - 1));
            for (int j = 1; j < i; j++){
                tmp.set(j, prev.get(j - 1)+prev.get(j));
            }
            rst.add(tmp);
        }
        return rst;
    }
}
